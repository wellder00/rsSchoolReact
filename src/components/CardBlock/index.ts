export { default as CardBlock } from './CardBlock';
