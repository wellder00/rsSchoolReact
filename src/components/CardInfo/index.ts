export { default as CardInfo } from './CardInfo';
