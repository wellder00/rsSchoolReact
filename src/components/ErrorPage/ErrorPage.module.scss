@import '../../styles/variables.scss';

.errorWrap {
  height: 100vh;
  text-align: center;
  background: $bg-main-color;
  font-size: 24px;
  color: white;
}

.title {
  margin-top: 0;
  padding-top: 20px;
}
