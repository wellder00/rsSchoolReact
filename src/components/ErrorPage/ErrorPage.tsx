import styles from './ErrorPage.module.scss';

const ErrorPage = () => {
  return (
    <div className={styles.errorWrap}>
      <h1 className={styles.title}>Oops!</h1>
      <p>Sorry, an unexpected error has occurred.</p>
      <p>Error</p>
    </div>
  );
};

export default ErrorPage;
