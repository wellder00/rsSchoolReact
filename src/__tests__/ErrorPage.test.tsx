import { render, screen } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { ErrorPage } from '../components/ErrorPage';
import React from 'react';

describe('Error page component', () => {
  it('Check page', () => {
    const history: MemoryHistory = createMemoryHistory({ initialEntries: ['/error'] });

    render(
      <MemoryRouter>
        <Routes>
          <Route path="/error" element={React.cloneElement(<ErrorPage />, { history })} />
        </Routes>
      </MemoryRouter>
    );

    expect(screen.getByRole('generic')).toBeInTheDocument();
  });
  it('Show error page', () => {
    const { asFragment } = render(<ErrorPage />);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText(/Oops!/i)).toBeInTheDocument();
  });
});
