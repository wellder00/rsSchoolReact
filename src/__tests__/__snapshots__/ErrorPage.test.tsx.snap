// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error page component > Show error page 1`] = `
<DocumentFragment>
  <div
    class="_errorWrap_8c5621"
  >
    <h1
      class="_title_8c5621"
    >
      Oops!
    </h1>
    <p>
      Sorry, an unexpected error has occurred.
    </p>
    <p>
      Error
    </p>
  </div>
</DocumentFragment>
`;
